spring-boot/proxy/src/test/java/security/ApiKeyAuthTokenVerifierTest.java

```diff
@@ -0,0 +1,189 @@
+        extendApiKeyFetcherMock_SetExpectedResponse(ResultOfApiKeyRequestToAdmin.EXCEPTION);
+        extendApiKeyManagerMock_ShallDeadlineBeExceeded(false);
+        extendApiKeyManagerMock_ShallApiKeyBeFoundInLocalApiKeyList(false);
+
+        assertNull(apiKeyAuthTokenVerifier.authenticate(apiKeyAuthToken));
+    }
+
+    @Test
+    public void apiKeyAuthorizedByLocalApiKeyListAndShouldBeAuthorized() throws ApiKeyFetchingFailedException {
+        extendApiKeyFetcherMock_SetExpectedResponse(ResultOfApiKeyRequestToAdmin.EXCEPTION);
+        extendApiKeyManagerMock_ShallDeadlineBeExceeded(false);
+        extendApiKeyManagerMock_ShallApiKeyBeFoundInLocalApiKeyList(true);
+        extendApiKeyVerifierMock_ShallApiKeyBeValid(true);
+
+        assertTrue(apiKeyAuthTokenVerifier.authenticate(apiKeyAuthToken).isAuthenticated());
+    }
+
+    @Test
+    public void exceedingTheDeadlineShouldRejectRequest() throws ApiKeyFetchingFailedException {
+        extendApiKeyFetcherMock_SetExpectedResponse(ResultOfApiKeyRequestToAdmin.EXCEPTION);
+        extendApiKeyManagerMock_ShallDeadlineBeExceeded(true);
+
+        assertNull(apiKeyAuthTokenVerifier.authenticate(apiKeyAuthToken));
+    }
+
+    @Test
+    public void apiKeyNotAuthorizedByLocalApiKeyListAndShouldNotBeAuthorized() throws ApiKeyFetchingFailedException {
+        extendApiKeyFetcherMock_SetExpectedResponse(ResultOfApiKeyRequestToAdmin.EXCEPTION);
+        extendApiKeyManagerMock_ShallDeadlineBeExceeded(false);
+        extendApiKeyManagerMock_ShallApiKeyBeFoundInLocalApiKeyList(true);
+        extendApiKeyVerifierMock_ShallApiKeyBeValid(false);
+
+        assertNull(apiKeyAuthTokenVerifier.authenticate(apiKeyAuthToken));
+    }
+
+
+    //Behavior of the mock objects
+
+    //apiKeyFetcherMock
+    private void extendApiKeyFetcherMock_SetExpectedResponse(ResultOfApiKeyRequestToAdmin resultOfApiKeyRequestToAdmin)
+            throws ApiKeyFetchingFailedException {
+        switch (resultOfApiKeyRequestToAdmin){
+            case NULL:
+                when(apiKeyFetcherMock.requestApiKeyFromAdmin(anyString())).thenReturn(null);
+                break;
+            case API_KEY:
+                when(apiKeyFetcherMock.requestApiKeyFromAdmin(apiKey.getKeyValue())).thenReturn(apiKey);
+                break;
+            case EXCEPTION:
+                when(apiKeyFetcherMock.requestApiKeyFromAdmin(anyString())).thenThrow(new ApiKeyFetchingFailedException());
+                break;
+        }
+    }
+
+    enum ResultOfApiKeyRequestToAdmin {
+        NULL, API_KEY, EXCEPTION
+    }
+
+    //apiKeyVerifierMock
+    private void extendApiKeyVerifierMock_ShallApiKeyBeValid(boolean shallApiKeyBeValid) {
+        when(apiKeyVerifierMock.isApiKeyValid(apiKey)).thenReturn(shallApiKeyBeValid);
+    }
+
+    //localApiKeyManagerMock
+    private void extendApiKeyManagerMock_ShallApiKeyBeFoundInLocalApiKeyList(boolean isFoundInLocalApiKeyList) {
```

Comment:
The word "extend" is a little bit confusing to me. Is that "mockito lingo"? If not, maybe call it `initLocalApiKeyManagerMock_...` ? or `setupLocalApiKeyManagerMock...`?

Reply:
Its not mockito lingo. My concerns about the terms 'setup' and 'initialize' are that all mocks are already initialized within the ```@BeforeEach```-setup method at the beginning. Probably I should change it to something like 'addBehaviorToApiKeyManager_...' which is what I actually meant with 'extend'.